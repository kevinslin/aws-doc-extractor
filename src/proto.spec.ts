import fs from 'fs';
import MarkdownIt from 'markdown-it';

describe.skip('proto', () => {

  it("extract", () => {
    const content = fs.readFileSync("data/sample_doc.md", 'utf-8');
    const md = new MarkdownIt();
    const tokens = md.parse(content, {});
    expect(tokens).toMatchSnapshot()
  });

});
