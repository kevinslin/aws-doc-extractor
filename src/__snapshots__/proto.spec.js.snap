// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proto extract 1`] = `
[
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      0,
      1,
    ],
    "markup": "##",
    "meta": null,
    "nesting": 1,
    "tag": "h2",
    "type": "heading_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "A header",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "A header",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": [
      0,
      1,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "##",
    "meta": null,
    "nesting": -1,
    "tag": "h2",
    "type": "heading_close",
  },
  
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      2,
      3,
    ],
    "markup": "",
    "meta": null,
    "nesting": 1,
    "tag": "p",
    "type": "paragraph_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "A paragraph",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "A paragraph",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": [
      2,
      3,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": -1,
    "tag": "p",
    "type": "paragraph_close",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      4,
      6,
    ],
    "markup": "",
    "meta": null,
    "nesting": 1,
    "tag": "p",
    "type": "paragraph_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "**",
        "meta": null,
        "nesting": 1,
        "tag": "strong",
        "type": "strong_open",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "Note",
        "hidden": false,
        "info": "",
        "level": 1,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "**",
        "meta": null,
        "nesting": -1,
        "tag": "strong",
        "type": "strong_close",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "br",
        "type": "hardbreak",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "In some cases, you may need to reboot your instance to provide permissions for the ",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "ec2-user",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "\`",
        "meta": null,
        "nesting": 0,
        "tag": "code",
        "type": "code_inline",
      },
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": " to access the Docker daemon. Try rebooting your instance if you see the following error:",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "**Note**  
In some cases, you may need to reboot your instance to provide permissions for the \`ec2-user\` to access the Docker daemon\\. Try rebooting your instance if you see the following error:",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": [
      4,
      6,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": -1,
    "tag": "p",
    "type": "paragraph_close",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "Cannot connect to the Docker daemon. Is the docker daemon running on this host?
",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      7,
      10,
    ],
    "markup": "\`\`\`",
    "meta": null,
    "nesting": 0,
    "tag": "code",
    "type": "fence",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      11,
      12,
    ],
    "markup": "##",
    "meta": null,
    "nesting": 1,
    "tag": "h2",
    "type": "heading_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "Create a Docker image<a name="create-container-image-create-image"></a>",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "Create a Docker image<a name="create-container-image-create-image"></a>",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": [
      11,
      12,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "##",
    "meta": null,
    "nesting": -1,
    "tag": "h2",
    "type": "heading_close",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": [
      13,
      14,
    ],
    "markup": "",
    "meta": null,
    "nesting": 1,
    "tag": "p",
    "type": "paragraph_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "Blah blah",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "Blah blah",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": [
      13,
      14,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": -1,
    "tag": "p",
    "type": "paragraph_close",
  },
]
`;
